src/ngAnimate/animateCss.js

```diff
@@ -292,14 +293,14 @@ function truthyTimingValue(val) {
   return val === 0 || val != null;
 }
 
-function getCssTransitionDurationStyle(duration, applyOnlyDuration) {
+function getCssTransitionStyle(timings, duration) {
   var style = TRANSITION_PROP;
   var value = duration + 's';
-  if (applyOnlyDuration) {
-    style += DURATION_KEY;
-  } else {
-    value += ' linear all';
-  }
+
+  value += ' ' + timings[TRANSITION_TIMING_PROP];
+  value += ' ' + timings[TRANSITION_PROPERTY_PROP];
+  value += timings[TRANSITION_DELAY_PROP] ? ' ' + timings[TRANSITION_DELAY_PROP] + 's' : '';
```

Comment:
Should these 3 lines use the keys of `DETECT_CSS_PROPERTIES` instead of the values ?
E.g. `timings.transitionDelay` instead of `timings[TRANSITIN_DELAY_PROP]` etc ?